src/lib.rs

```diff
@@ -281,30 +281,52 @@ impl Storage for S3Storage {
             .map_err(|e| zerror!("Get operation failed: {e}"))?
             .map_err(|e| zerror!("Get operation failed: {e}"))?;
 
-        let futures = objects.into_iter().map(|object| {
+        let path_prefix = self.config.path_prefix.to_owned();
+        let futures = objects.into_iter().filter_map(|object| {
+            let object_key = match object.key() {
+                Some(key) => key.to_string(),
+                None => {
+                    log::error!("Could not get key for object {:?}", object);
+                    return None;
+                }
+            };
+
+            if object_key == NONE_KEY {
+                return None;
+            }
+
+            let key = path_prefix
+                .as_ref()
+                .map_or(object_key.to_string(), |prefix| {
```

Comment:
suggestion: perhaps this could use fewer clones if the whole block was turned into 
```
path_prefix.as_ref().unwrap_or_default() + object_key
```
as that would avoid cloning `object_key` potentially twice?